src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts

```diff
@@ -475,26 +477,49 @@ export class SQLFuture extends Disposable implements FutureInternal {
 
 	private async processResultSets(batch: BatchSummary): Promise<void> {
 		try {
+			let queryRowsPromises: Promise<void>[] = [];
 			for (let resultSet of batch.resultSetSummaries) {
 				let rowCount = resultSet.rowCount > this.configuredMaxRows ? this.configuredMaxRows : resultSet.rowCount;
 				if (rowCount === this.configuredMaxRows) {
 					this.handleMessage(localize('sqlMaxRowsDisplayed', "Displaying Top {0} rows.", rowCount));
 				}
-				await this.sendResultSetAsIOPub(rowCount, resultSet);
+				queryRowsPromises.push(this.getAllQueryRows(rowCount, resultSet));
+			}
+			// We want to display table in the same order
```

Comment:
Promise.allSettled()?

Reply:
Yeah agreed, it'd be nicer to wait on all and also handle the case where a single result set promise fails cleanly.

Reply:
I'm a little confused here. If we waited on everything then we're negating the perf improvement by waiting for all queries to finish, right?

In addition if, for example, the 2nd query in a batch fails, don't we still want to output results from the 1st query? This is what the query editor does.